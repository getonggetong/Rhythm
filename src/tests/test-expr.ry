/*	Expression Test

	Tests the following:
	1. Operator associativity and precedence
	2. All operator functionality with a statement
	3. Assignments
*/

init() {
	print(1 + 2 - 3);
	a = [A4,B4,C4];
	b = A4 - (1 + 2) * 4 / 2 :: [[a[1]+1,B4]] -> 3 >> 3 << 1;
	print(b);
	b++;
	print(b);
	b--;
	print(b);
	b >>= 2;
	print(b);
	b <<= 3;
	print(b);
	b += 1;
	print(b);
	b -= 2;
	print(b);
	b ::= G#7;
	print(b);
	b *= 2;
	print(b);
	b /= 4;
	print(b);
}