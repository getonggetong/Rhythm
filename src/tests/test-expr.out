0
[F#3.2,[C6.4,B6.4],[C6.4,B6.4],[C6.4,B6.4]]
[G3.2,[C#6.4,C6.4],[C#6.4,C6.4],[C#6.4,C6.4]]
[F#3.2,[C6.4,B6.4],[C6.4,B6.4],[C6.4,B6.4]]
[F#5.2,[C8.4,B8.4],[C8.4,B8.4],[C8.4,B8.4]]
[F#2.2,[C5.4,B5.4],[C5.4,B5.4],[C5.4,B5.4]]
[G2.2,[C#5.4,C5.4],[C#5.4,C5.4],[C#5.4,C5.4]]
[F2.2,[B5.4,A#5.4],[B5.4,A#5.4],[B5.4,A#5.4]]
[F2.2,[B5.4,A#5.4],[B5.4,A#5.4],[B5.4,A#5.4],G#7]
[F2.1,[B5.2,A#5.2],[B5.2,A#5.2],[B5.2,A#5.2],G#7.2]
[F2.4,[B5.8,A#5.8],[B5.8,A#5.8],[B5.8,A#5.8],G#7.8]
